### PR TITLE
Reproducible development environment with docker-compose

### PR DESCRIPTION
**Pull request by jehrhardt**
The project has many dependencies during development. While libraries and most tools are managed via npm, bower or bundler, other dependencies are not like:

* Ruby
* Node.js
* Postgres

Depending on the operating system of the developer, these installations might be different and the versions as well. This introduces problems from time to time and the solution is to introduce an unified and completely automated way to install and run everything in a reproducible way.

This pull request provides such a reproducible way using Docker. All developers will run the application in exactly the same environment, which is very similar to the production system. This reduces differences between environments as much as possible. A developer does not need to install anything on her computer than [Docker Community Edition](https://store.docker.com/search?offering=community&type=edition). Ensure installation with the following command

```sh
docker-compose --version
```

Once Docker has been installed, creating the local environment from scratch is done by the script

```sh
dev/reset
```

This will create the local environment. Once the environment is set up the application can be started

```sh
dev/start
```

Of course it is available under http://localhost:4200 as before.

By running the `dev/reset` command again, the complete local environment is deleted and recreated from scratch. This makes resolving issues based on the infrastructure very simple 😉.

All changes in this pull request are compatible to the previous set up. Thus Docker is an add-on, that can be tried by all developers without any breaks.

### Future improvements
* A documentation in the README is missing (should follow)
* Only backend, frontend and the database are running in Docker, tests are not integrated yet
* The Travis CI build should run the same Docker containers (same environment even on Travis CI)
* The reset command currently only runs the database migration, but does not run `db:seed` to get the test data loaded

### CODE REVIEW

**Comment by roschaefer** [comment]
This is awesome :balloon: I tried it on my local machine and everything works like a charm :dizzy: We can merge this PR as soon as we find and fix the problem the current broken build on master.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    environment:
+      - DATABASE_URL=postgresql://db
+      - POSTGRES_USERNAME=postgres
+    links:
+      - db
+    command: bin/rails s -b 0.0.0.0
+  db:
+    image: postgres:9.2-alpine
+    environment:
+      - POSTGRES_DB=rundfunk-backend-development
```
Comment (roschaefer) [inline docker-compose.yml]: that's the reason why we can't run tests at the moment?

Comment (jehrhardt) [inline docker-compose.yml]: No, there is no issue with tests. It just needs to be done 😉.

I wanted to wait until everything works and everybody feels comfortable with it.

**Comment by roschaefer** [comment]
This build is broken for probably the same reason like our master is broken. There should be no problem with this PR: https://github.com/roschaefer/rundfunk-mitbestimmen/issues/138

**Pull request merged**